pkg/plugins/policies/meshtimeout/plugin/v1alpha1/plugin.go

```diff
@@ -96,15 +103,26 @@ func applyToOutbounds(rules core_xds.ToRules, outboundListeners map[mesh_proto.O
 			continue
 		}
 
+		clustersToConfigure := []*envoy_cluster.Cluster{}
 		serviceName := outbound.GetTagsIncludingLegacy()[mesh_proto.ServiceTag]
+		splitCluster, ok := outboundSplitClusters[serviceName]
+		if ok {
+			clustersToConfigure = append(clustersToConfigure, splitCluster...)
+		}
 		cluster, ok := outboundClusters[serviceName]
-		if !ok {
+		if ok {
+			clustersToConfigure = append(clustersToConfigure, cluster)
+		}
```

Comment:
I think you should be able to use `GatherTargetedClusters` here too? no?

Reply:
Here we match by the service name, and in `GatherTargetedClusters` we have `cluster` to `serviceName` so I don't think we can.

Reply:
ok, let's figure it out later